concourse/templates/default.yaml

```diff
@@ -162,18 +161,21 @@ source_repo = job_variant.main_repository()
 % if job_step.image():
 <%
 image_reference, tag = job_step.image().split(':', 1)
+## We have been given an image. Depending on whether a registry name is also configured or not we
+## set the proper registry config and render all required credentials explicitly
+if job_step.registry():
+  container_registry = config_set.container_registry(job_step.registry())
```

Comment:
why not just use config_set's default mechanism? if no cfg_element name is given, the default element is returned. this should also work if None is passed.